command/agent/local.go

```diff
@@ -379,6 +379,12 @@ func (l *localState) setSyncState() error {
 		}
 
 		// If our definition is different, we need to update it
+		if existing.EnableTagDrift {
+			l.logger.Printf("[DEBUG] Tag drift enabled.")
+			existing.Tags = service.Tags
```

Comment:
@slackpad if we are doing a `DeepEqual`, then you are right, the definitions have to be perfectly equivalent. What I was suggesting was having like a `ServiceEquals(s1, s2 *structs.NodeService, drift bool) bool` kind of function, which would just skip the tag check if drift is enabled.

You're right about the local state, my bad.

One other thing though - If we are modifying the local agent state then we aren't enabling drift, we are instead flipping it so that the remote state always wins, unless I'm missing something else here.